addon/mixins/adapter-fetch.js

```diff
@@ -109,6 +109,23 @@ export function mungOptionsForFetch(_options, adapter) {
 
   return options;
 }
+/**
+ * Function checks that there is a body in the response. Otherwise JSON.parse will throw an error.
+ * If there is not, return an empty object and let Ember data handle the empty response
+ * @param {Response} response
+ * @returns {Promise}
+ */
+export function determineBodyPromise(response) {
+  let bodyPromise;
+  const contentLength = response.headers.get('content-length');
+
+  if (contentLength && Number(contentLength)) {
```

Comment:
sorry to nit-pick curious about the following possible values of `Number(contentLength)`

* `0` empty payload seems good
* '-1` `response.json()` may be unhappy

I suspect the if statement should be `if (contentLenNumber(contentLength) > 0) { ... }`